test/varmat_compatibility_summary.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/python
+
+import argparse
+import json
+import sys
+
+from argparse import ArgumentDefaultsHelpFormatter, ArgumentParser
+from signature_parser import SignatureParser
+
+def convert_signatures_list_to_functions(signatures):
+    """
+    Take a list of signatures and extract the unique set of functions they represent
+
+    :param signatures: List of signatures
+    """
+    functions = set()
+    for signature in signatures:
+        sp = SignatureParser(signature)
+        functions.add(sp.function_name)
+    return functions
+
+def select_signatures_matching_functions(signatures, functions):
+    """
+    Select from a collection of signatures those which have function names in the given function list
+
+    :param signatures: List of signatures
+    :param functions: Function names to match against
+    """
+    signatures_out = set()
+    for signature in signatures:
+        sp = SignatureParser(signature)
+
+        if sp.function_name not in functions:
+            continue
+
+        signatures_out.add(signature)
+    return signatures_out
+
+def remove_signatures_matching_functions(signatures, functions):
+    """
+    Filter from a collection of signatures those which have function names in the given function list
+
+    :param signatures: List of signatures
+    :param functions: Function names to filter against
+    """
+    signatures_out = set()
+    for signature in signatures:
+        sp = SignatureParser(signature)
+
+        if sp.function_name in functions:
+            continue
+
+        signatures_out.add(signature)
+    return signatures_out
+
+def process_results(results, functions, which, fully, names):
+    """
+    Processes results of varmat compatiblity results file and returns list of functions/signatures that
+    match the given criteria
+
+    :param results: Results of varmat compatibility calculation
+    :param functions: List of function names to check compatibility for
+    :param which: For which type of compatibility should functions be printed
+    :param fully: Only print fully compatible/incompatible signatures (no partial varmat support)
```

Comment:
partial support makes sense only for functions (and not for signatures), right?

Reply:
I can reword this. It only prints signatures if the function that signature is associated with is fully supported, so it makes sense and is useful.

Reply:
That was just a question. No need to change anything here.

Reply:
Unfortunately your feedback has already been used to improve the docs. No takesies backsies